core/src/fields.rs

```diff
@@ -512,15 +512,22 @@ impl MongoFields {
         loop {
             if self.collections_for_db.is_some() {
                 for current_collection in self.collections_for_db.as_mut().unwrap() {
-                    let get_schema_cmd = doc! {"sqlGetSchema": current_collection.unwrap().name};
+                    let collection_name = current_collection.unwrap().name;
+                    let get_schema_cmd = doc! {"sqlGetSchema": collection_name.clone()};
 
                     let db = mongo_connection.client.database(&self.current_db_name);
-                    let current_col_metadata_response: SqlGetSchemaResponse =
+                    let current_col_metadata_response: Result<SqlGetSchemaResponse> =
                         bson::from_document(db.run_command(get_schema_cmd, None).unwrap())
-                            .map_err(Error::BsonDeserialization)
-                            .unwrap();
+                            .map_err(Error::BsonDeserialization);
+                    if current_col_metadata_response.is_err() {
```

Comment:
I think it's best to just skip a collection if we fail to deserialize the response, rather than killing iteration.

Reply:
It would fail silently, but I think that's the best thing to do if somehow there is a junk response from sqlGetSchema.

Reply:
An alternative to skipping could be to mark the collection as having the "Any" schema. Although that would just defer the problem to other parts of the code, unless we handle this case all the way to the end and report columns from such problematic collections as having "BSON" (unknown) as the datatype.

Yea, I'm convinced skipping (for now) is the best move.

Reply:
Right now it will only fail to deserialize if the schema isn't an object. If it's not an object we don't have columns to report anyway.